extras/src/main/java/com/google/gson/graph/GraphAdapterBuilder.java

```diff
@@ -111,20 +110,17 @@ public void registerOn(GsonBuilder gsonBuilder) {
            *     out the object's value as a part of #1.
            */
 
-          if (graph == null) {
-            writeEntireGraph = true;
+          if (graph == null) { 
             graph = new Graph(new IdentityHashMap<Object, Element<?>>());
-          }
 
-          @SuppressWarnings("unchecked") // graph.map guarantees consistency between value and T
-          Element<T> element = (Element<T>) graph.map.get(value);
-          if (element == null) {
-            element = new Element<T>(value, graph.nextName(), typeAdapter, null);
-            graph.map.put(value, element);
-            graph.queue.add(element);
-          }
+            @SuppressWarnings("unchecked") // graph.map guarantees consistency between value and T
+            Element<T> element = (Element<T>) graph.map.get(value);
```

Comment:
This will always be null since nothing has been written to the IdentityHashMap from the previous statement. So you can simplify here.

Reply:
You're right, i'll do that